### PR TITLE
feat(engine): make new payload canonical action

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Builds on top of https://github.com/paradigmxyz/reth/pull/8041.

Introduces `MakeNewPayloadCanonical` action which is triggered when new payload is valid and current FCU head.

### CODE REVIEW

**Comment by rkrasiuk** [comment]
hive run https://github.com/paradigmxyz/reth/actions/runs/8921583725

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -1703,37 +1703,47 @@ where
...
-                                        validation_error: error.to_string(),
-                                    },
-                                    // TODO: return a proper latest valid hash
-                                    // See: <https://github.com/paradigmxyz/reth/issues/7146>
-                                    self.forkchoice_state_tracker.last_valid_head(),
-                                ));
-                                let _ = tx.send(status);
-                                return Ok(EngineEventOutcome::Processed)
-                            }
+                            let previous_action = self.blockchain_tree_action.replace(
```
Comment (mattsse) [inline crates/consensus/beacon/src/engine/mod.rs:1706]: I assume eventually this will be converted into a Vec or channel?

**Review by mattsse — APPROVED** [review]

File: crates/consensus/beacon/src/engine/mod.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/consensus/beacon/src/engine/mod.rs:1706]: exactly! but baby steps for now

**Review by shekhirin — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]
might clash with #8044 a bit cc @fgimenez

**Pull request merged**